### PR TITLE
restructure communication section

### PR DESCRIPTION
**Pull request by esc**
Add a section pointing to our discourse and move the mailing-list down
to emphasise that it can still be used but is no longer the preferred
method of communication.

### CODE REVIEW

**Comment by esc** [comment]
Since we introduced this recently and already have a badge in the README, it's worth adding this to the dev docs for the next release.

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -38,6 +28,22 @@ developers to keep up with.
...
+discussions and roadmap planning. There are various categories available and it
+can reached at: `numba.discourse.group https://numba.discourse.group/`.
+
+Mailing-list
+''''''''''''
+
+We have a public mailing-list that you can e-mail at numba-users@anaconda.com.
+ You can subscribe and read the archives on
+`Google Groups <https://groups.google.com/a/continuum.io/forum/#!forum/numba-users>`_,
+and there is also a `Gmane mirror <http://news.gmane.org/gmane.comp.python.numba.user>`_
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst:44]: Don't think the gmane mirror is up any more?

Comment (esc) [reply docs/source/developer/contributing.rst]: no it isn't and as a result, I have removed the link to it.

File: docs/source/developer/contributing.rst [hunk 2]
```diff
@@ -38,6 +28,20 @@ developers to keep up with.
 Note that the Github issue tracker is the best place to report bugs.  Bug
 reports in chat are difficult to track and likely to be lost.
 
+Forum
+.....
+
+Numba uses Discourse as a forum for longer running threads such as design
+discussions and roadmap planning. There are various categories available and it
+can reached at: `numba.discourse.group https://numba.discourse.group/`_.
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst:36]:
```suggestion
can reached at: `numba.discourse.group <https://numba.discourse.group/>`_.
```

File: docs/source/developer/contributing.rst [hunk 3]
```diff
@@ -38,6 +28,20 @@ developers to keep up with.
...
+
+Mailing-list
+''''''''''''
+
+We have a public mailing-list that you can e-mail at numba-users@anaconda.com.
+ You can subscribe and read the archives on
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst:42]:
```suggestion
You can subscribe and read the archives on
```

File: docs/source/developer/contributing.rst (hunk 2 above)
Comment (esc) [reply docs/source/developer/contributing.rst:36]: I fixed the link already. However, additionally there is a 'be'  missing here.

Comment (stuartarchibald) [reply docs/source/developer/contributing.rst]: build is still failing, I'm pretty sure this syntax is needed?!

Comment (esc) [reply docs/source/developer/contributing.rst]: Indeed.

**Pull request merged**